systemctl_mqtt/__init__.py

```diff
@@ -266,14 +291,21 @@ def trigger(self, state: _State) -> None:
     "suspend": _MQTTActionSuspend(),
 }
 
-
 async def _mqtt_message_loop(*, state: _State, mqtt_client: aiomqtt.Client) -> None:
     action_by_topic: typing.Dict[str, _MQTTAction] = {}
     for topic_suffix, action in _MQTT_TOPIC_SUFFIX_ACTION_MAPPING.items():
         topic = state.mqtt_topic_prefix + "/" + topic_suffix
         _LOGGER.info("subscribing to %s", topic)
         await mqtt_client.subscribe(topic)
         action_by_topic[topic] = action
+
+    for unit_name in state.controlled_system_unit_names:
+        topic = state.mqtt_topic_prefix + "/unit/system/" + unit_name + "/restart"
+        _LOGGER.info("subscribing to %s", topic)
+        await mqtt_client.subscribe(topic)
+        action = _MQTTActionControlUnit()
```

Comment:
Loop looks good for now. Maybe will refactor it later to match by wildcard "+".

But I'll need to pass the unit name to the action for reference in `trigger`:
```suggestion
        action = _MQTTActionControlUnit(unit_name=unit_name)
```

```
class _MQTTActionControlUnit(_MQTTAction):
    def __init__(self, unit_name: str):
        self._unit_name = unit_name
    def trigger(self, state: _State) -> None:
        systemctl_mqtt._dbus.control_manager.control_unit(unit_name=self._unit_name)

```

Reply:
wow, this was the missing piece - i cannot thank you enough!